h2/utilities.py

```diff
@@ -291,29 +295,101 @@ def _check_host_authority_header(headers, hdr_validation_flags):
     host_header_val = None
 
     for header in headers:
-        if header[0] == b':authority':
+        if header[0] in (b':authority', u':authority'):
```

Comment:
This shouldn't actually be needed here because received header blocks should always be bytes: they only get optionally decoded after validation. However, given that this loop is now basically identical to the one in `_check_send_authority_host_header`, I recommend hoisting this loop out to `_validate_host_authority_header` too. Let's duplicate as little code as possible.